core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -158,6 +158,11 @@ class GroupMetadataManager(brokerId: Int,
 
   def isLoading(): Boolean = inLock(partitionLock) { loadingPartitions.nonEmpty }
 
+  // return true iff group is owned and the group doesn't exist
+  def groupNotExists(groupId: String) = inLock(partitionLock) {
+    isGroupLocal(groupId) && (!groupMetadataCache.contains(groupId) || groupMetadataCache.get(groupId).is(Dead))
```

Comment:
Should have mentioned before, but we do need to grab the group lock to check the state.

Reply:
Correct, thanks for catching. Hopefully the new commit works.